esp/public/media/scripts/program/modules/scheduling_checks.jsx

```diff
@@ -114,3 +148,71 @@ var RefreshButton = React.createClass({
     </button>;
   },
 });
+
+
+// Modified from react-json-table example code.
+// Required props: rows, saveState (with sort and greyed attributes), settings (with header attribute)
+var SelectTable = React.createClass({
+  getInitialState: function(){
+    // We will store the sorted column and whether each row is greyed out
+    return {sort: this.props.saveState.sort, greyed : this.props.saveState.greyed};
+  },  
+  render: function(){
+    var me = this,
+        // clone the rows
+        items = this.props.rows.slice()
+    ;
+    // Sort the table
+    if( this.state.sort ){
+      items.sort( function( a, b ){
+         return a[ me.state.sort ] > b[ me.state.sort ] ? 1 : -1;
+      });
+    }
+      
+    if (this.props.columns == undefined) {
+      return <JsonTable 
+        rows={items} 
+        settings={ this.getSettings() } 
+        onClickHeader={ this.onClickHeader }
+        onClickRow={ this.onClickRow }
+      />;
+    } else {
+      return <JsonTable 
+        rows={items} 
+        columns={this.props.columns}
+        settings={ this.getSettings() } 
+        onClickHeader={ this.onClickHeader }
+        onClickRow={ this.onClickRow }
+      />;
+    }
+  },
+  
+  getSettings: function(){
+      var me = this;
+      // We will add some classes to the selected rows and cells
+      return {
+        headerClass: function( current, key ){
+            if( me.state.sort == key )
+              return current + ' headerSelected';
+            return current;
+        },
+        rowClass: function( current, item ){
+          if( me.state.greyed[item] )
+            return current + ' rowGreyed';
+          return current;
+        },
+        header: this.props.settings.header
+      };
+  },
+    
+  onClickHeader: function( e, column ){
+    this.props.saveState.sort = column;
```

Comment:
The current operation is that modifying the prop will also modify the check's saveState, but I'll find a different mechanism if this isn't good practice.